### PR TITLE
just load the key from the repo to avoid apt-key issues

### PR DESCRIPTION
**Pull request by jvonau**
### Fixes bug:
known intermittent issues connecting to keyserver.ubuntu.com
### Feature
allows for easier container building

### CODE REVIEW

**Comment by tim-moody** [comment]
makes sense to me

**Comment by holta** [comment]
It's even more serious: "apt-key will last be available in Debian 11 and Ubuntu 22.04."

Article below (also) says keys should be stored in `/usr/share/keyrings/<myrepository>-archive-keyring.gpg` as adding them to `/etc/apt/trusted.gpg.d` is insecure (basically this is the reason command apt-key is going away).

> apt-key Is Deprecated. How To Add OpenPGP Repository Signing Keys Without It On Debian, Ubuntu, Linux Mint, Pop!_OS, Etc.
https://www.linuxuprising.com/2021/01/apt-key-is-deprecated-how-to-add.html

I'll work on it later tonight.

**Comment by jvonau** [comment]
This hack works on Ubuntu, in place of apt-key
>add-apt-repository -y  ppa:ansible/ansible
rm /etc/apt/sources.list.d/ansible-ubuntu-ansible-*.list

The issue with /usr/share/keyrings/<myrepository>-archive-keyring.gpg is that should be what apt-key or add-apt-repository should do for the user, but doesn't. Until the upstream plumbing is figured out to replace /etc/apt/trusted.gpg.d/ don't try to over-engineer the fix today and use 'KISS'.

**Comment by jvonau** [comment]
I tried to find a web source to document where the key lives and verify but was unsuccessful in the search, that would be the component needed to use /usr/share/keyrings/. I used the one supplied by add-apt-repository as the source for this PR if there is any questions.

**Comment by jvonau** [comment]
https://askubuntu.com/questions/1286545/what-commands-exactly-should-replace-the-deprecated-apt-key/1307181#1307181
Looks like iiab-ansible.list just needs to reference /usr/local/share/keyrings/ I'll update the PR

**Comment by holta** [comment]
FWIW the directory `/usr/share/keyrings` exists on all relevant OS's (RaspiOS 10+, Ubuntu 20+, and Debian 10+).

So we could certainly use that (instead of `/usr/local/share/keyrings`) if using such an approach.

**Comment by jvonau** [comment]
> FWIW the directory `/usr/share/keyrings` exists on all relevant OS's (RaspiOS 10+, Ubuntu 20+, and Debian 10+).
> 
> So we could certainly use that (instead of `/usr/local/share/keyrings`) if using such an approach.

From the above link
>There's nothing special about that location, it's just convention that /usr/local is for stuff that's specific to this machine, share because it's not a binary or a library or specific to any given user, and keyrings is just a descriptive name.

>ubuntu@box:~$ ls /usr/share/keyrings/
ubuntu-advantage-cis.gpg               ubuntu-archive-removed-keys.gpg
ubuntu-advantage-esm-apps.gpg          ubuntu-cloudimage-keyring.gpg
ubuntu-advantage-esm-infra-trusty.gpg  ubuntu-cloudimage-removed-keys.gpg
ubuntu-advantage-fips.gpg              ubuntu-master-keyring.gpg
ubuntu-archive-keyring.gpg

Lets keep /usr/share/keyrings clear of any 'third party additions' (us) and keep that for distro provided files until we know how upstream ansible along with the rest of the repos that might get enabled along the way are going to handle this change in the future, this is the most foolproof method today IMHO

**Comment by jvonau** [comment]
Or just point to where the key lives and be done with it..

**Comment by holta** [comment]
> how upstream ansible along with the rest of the repos...are going to handle this change

No question Ansible hasn't yet got their act together here, on their install page that's often out-of-date and incorrect: https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#installing-ansible-on-debian

_(Imagine...one day installing Ansible might actually be Dead Simple!  Ok, maybe not for a few more years, hah :-)_

Old excerpt from https://github.com/iiab/iiab/blob/master/scripts/ansible#L100-L104:
```
# In future we might instead consider 'add-apt-repository ppa:ansible/ansible'
# or 'apt-add-repository ppa:ansible/bionic/ansible' etc, e.g. for streamlined
# removal using 'apt-add-repository -r' -- however that currently requires
# 'apt install software-properties-common' which drags in a dozen packages we
# might not want, e.g. unattended-upgrades, packagekit etc.
```

Refs: #2721, PR #2742, PR #2743

File: scripts/ansible [hunk 1]
```diff
@@ -110,8 +112,8 @@ echo "deb http://ppa.launchpad.net/ansible/ansible/ubuntu $CODENAME main" \
 #echo "deb http://ppa.launchpad.net/ansible/ansible/ubuntu disco main" \
     #    > /etc/apt/sources.list.d/iiab-ansible.list
 
-echo -e '\nIF YOU FACE ERROR "signatures couldn'"'"'t be verified because the public key is not available" THEN REPEATEDLY RE-RUN "sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 93C4A3FD7BB9C367"\n'
-apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 93C4A3FD7BB9C367
+#echo -e '\nIF YOU FACE ERROR "signatures couldn'"'"'t be verified because the public key is not available" THEN REPEATEDLY RE-RUN "sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 93C4A3FD7BB9C367"\n'
+#apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 93C4A3FD7BB9C367
 echo -e 'PPA source "deb http://ppa.launchpad.net/ansible/ansible/ubuntu '$CODENAME' main"'
```
Comment (holta) [inline scripts/ansible:118]: Eventually we should make this Line 117...correspond to Lines 100-101 that it is documenting.

Comment (jvonau) [reply scripts/ansible:118]: Think the change should stay the way it is to denote a change in method of **using** the gpg key with signed-by= as opposed to the earlier method of trying to **install** the key. Keeps the commented code in one place

**Comment by jvonau** [comment]
> Old excerpt from https://github.com/iiab/iiab/blob/master/scripts/ansible#L100-L104:
> 
Old? The comment is still present just lower, have a better look. Juggled the notes around a bit

**Comment by holta** [comment]
FYI @deldesir in Haiti told me he'll try to make time to test this PR over the wkd, likely on Ubuntu Server 21.04 and Ubuntu Desktop 21.04 VM's for starters?  Debian 11 RC1 and 64-bit RaspiOS (etc) also being possibilities (maybe) as he's worked on such in the past.

(Possibly rolled up with PR #2742 or PR #2743 somehow, given ansible-core 2.11's release is now imminent...on Monday in 3 days from now.)

Refs: #2744 #2748 #2749

**Comment by jvonau** [comment]
> FYI deldesir in Haiti told me he'll try to make time to test this PR over the wkd, likely on Ubuntu Server 21.04 and Ubuntu Desktop 21.04 VM's for starters? 

@holta How about you take the time to test this PR yourself as you seem to think you in charge are QA

**Comment by holta** [comment]
> @holta How about you take the time to test this PR yourself as you seem to think you in charge are QA

Yes I'll try to work with @deldesir over the wkd, if we can hopefully align our schedules.  FYI he has a job and attends church Friday nights, so is not available today.

**Comment by holta** [comment]
Thank you to @jvonau and @deldesir who both greatly helped make this research & testing happen.

See PR #2750 which should be merged in coming hours, building on this PR.

**Comment by jvonau** [comment]
>ubuntu@box:~/$  ls -la /usr/share/keyrings/*.gpg 
-rw-r--r-- 1 root root 2236 Sep  1  2020 /usr/share/keyrings/ubuntu-advantage-esm-apps.gpg
-rw-r--r-- 1 root root 2264 Sep  1  2020 /usr/share/keyrings/ubuntu-advantage-esm-infra-trusty.gpg
-rw-r--r-- 1 root root 2275 Sep  1  2020 /usr/share/keyrings/ubuntu-advantage-fips.gpg
-rw-r--r-- 1 root root 7399 Sep 17  2018 /usr/share/keyrings/ubuntu-archive-keyring.gpg
-rw-r--r-- 1 root root 6713 Oct 27  2016 /usr/share/keyrings/ubuntu-archive-removed-keys.gpg
-rw-r--r-- 1 root root 4097 Feb  6  2018 /usr/share/keyrings/ubuntu-cloudimage-keyring.gpg
-rw-r--r-- 1 root root    0 Jan 17  2018 /usr/share/keyrings/ubuntu-cloudimage-removed-keys.gpg
-rw-r--r-- 1 root root 1227 May 27  2010 /usr/share/keyrings/ubuntu-master-keyring.gpg

> ubuntu@box:~/$ dpkg -S /usr/share/keyrings/*.gpg 
ubuntu-advantage-tools: /usr/share/keyrings/ubuntu-advantage-esm-apps.gpg
ubuntu-advantage-tools: /usr/share/keyrings/ubuntu-advantage-esm-infra-trusty.gpg
ubuntu-advantage-tools: /usr/share/keyrings/ubuntu-advantage-fips.gpg
ubuntu-keyring: /usr/share/keyrings/ubuntu-archive-keyring.gpg
ubuntu-keyring: /usr/share/keyrings/ubuntu-archive-removed-keys.gpg
ubuntu-keyring: /usr/share/keyrings/ubuntu-cloudimage-keyring.gpg
ubuntu-keyring: /usr/share/keyrings/ubuntu-cloudimage-removed-keys.gpg
ubuntu-keyring: /usr/share/keyrings/ubuntu-master-keyring.gpg

Gut feeling tells me /usr/share/keyrings/ is intended to be reserved for distro management, the left most entry is the deb package name that supplied the file. Bloggers don't always provide sound advice, more of a "this is how I did it" point of view.

**Comment by holta** [comment]
> Bloggers don't always provide sound advice, more of a "this is how I did it" point of view.

Agreed 100%.  Norms are evolving.

**Pull request closed without merging**